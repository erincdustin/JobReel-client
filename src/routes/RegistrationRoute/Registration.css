.registration-page {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 20px;
  background-attachment: scroll;
  position: relative;
}

.registration-image > img {
  float: left;
  height: 320px;
  width: 320px;
  position: absolute;
  left: -35%;
  z-index: -1;
}

.form-registration {
  grid-column-start: 6;
  grid-column-end: 9;
  border: 1px black solid;
  min-width: 300px;
  margin: 0 auto;
  padding: 5%;
  line-height: 150%;
  border-radius: 10px;
  text-align: center;
  height: 370px;
  background: #3b3a50;
  color: #eeebd9;
  box-shadow: 0.5em 0.5em 0.2em rgb(48, 39, 39);
}

.registration-description {
  grid-column-start: 1;
  grid-column-end: 5;
  height: 25em;
}

.error {
  color: red;
}

@media screen and (max-width: 960px) {
  .form-registration {
    grid-column-start: 2;
    grid-column-end: 8;
  }
}