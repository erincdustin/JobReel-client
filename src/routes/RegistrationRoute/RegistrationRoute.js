import React, { Component } from 'react'
import Description from '../../components/Description/Description'
import { Link } from 'react-router-dom'
import { Input, Required, Label } from '../../components/Form/Form'
import AuthApiService from '../../services/auth-api-service'
import Button from '../../components/Button/Button'



export default class RegistrationRoute extends Component {
    static defaultProps = {
        history: {
            push: () => { },
        },
    }

    state = { error: null }

    firstInput = React.createRef()

    handleSubmit = ev => {
        ev.preventDefault()
        const { email, first_name, last_name, username, password } = ev.target
        AuthApiService.postUser({
            email: email.value,
            first_name: first_name.value,
            last_name: last_name.value,
            username: username.value,
            password: password.value,
        })
            .then(() => {
                email.value = ''
                first_name.value = ''
                last_name.value = ''
                username.value = ''
                password.value = ''
                const { history } = this.props
                history.push('/login')
            })
            .catch(res => {
                this.setState({ error: res.error })
            })
    }

    componentDidMount() {
        this.firstInput.current.focus()
    }

    render() {
        const { error } = this.state
        return (
            <section className="authSection">
                <form id="form-container"
                    onSubmit={this.handleSubmit}
                >
                    <div role='alert'>
                        {error && <p>{error}</p>}
                    </div>
                    <div>
                        <Label htmlFor='registration-email-input'>
                            Email:<Required />
                        </Label>
                        <Input
                            ref={this.firstInput}
                            id='registration-email-input'
                            name='email'
                            required
                        />
                    </div>
                    <div>
                        <Label htmlFor='registration-first-name-input'>
                            First name:<Required />
                        </Label>
                        <Input

                            id='registration-first-name-input'
                            name='first-name'
                            required
                        />
                    </div>
                    <div>
                        <Label htmlFor='registration-last-name-input'>
                            Last name:<Required />
                        </Label>
                        <Input
                            id='registration-last-name-input'
                            name='last-name'
                            required
                        />
                    </div>
                <div>
                    <Label htmlFor='registration-username-input'>
                        Choose a username<Required />
                    </Label>
                    <Input
                        id='registration-username-input'
                        name='username'
                        required
                    />
                </div>
                <div>
                    <Label htmlFor='registration-password-input'>
                        Choose a password<Required />
                    </Label>
                    <Input
                        id='registration-password-input'
                        name='password'
                        type='password'
                        required
                    />
                </div>
                <div>
                    <Button type="submit">
                        Sign up
                        </Button>
                </div>
                {' '}
                <Link className="already" to='/login'>Already have an account?</Link>
                <Description />
                </form>
            </section >
        )
    }
}

