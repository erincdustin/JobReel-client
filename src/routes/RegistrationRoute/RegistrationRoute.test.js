import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import RegistrationRoute from './RegistrationRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('RegistrationRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><RegistrationRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

let focusOnTargetSpy = jest.fn();

jest
  .spyOn(RegistrationRoute.prototype, 'focusOnTarget')
  .mockImplementation(focusOnTargetSpy);


describe('RegistrationRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<RegistrationRoute />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})