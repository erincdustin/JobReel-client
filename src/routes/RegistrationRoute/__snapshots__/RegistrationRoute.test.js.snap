// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationRoute snapshot test renders the UI as expected 1`] = `
<div
  className="registration-page"
>
  <div
    className="registration-image"
  >
    <img
      alt="registration-background"
      src="registration.svg"
    />
  </div>
  <Description
    className="registration-description"
  />
  <div
    className="form-registration"
  >
    <form
      id="form-container"
      onSubmit={[Function]}
    >
      <div
        className="error-message"
        role="alert"
      />
      <div>
        <Label
          htmlFor="registration-email-input"
        >
          Email:
          <Required />
        </Label>
        <br />
        <ForwardRef
          id="registration-email-input"
          name="email"
          required={true}
        />
      </div>
      <div>
        <Label
          htmlFor="registration-first-name-input"
        >
          First name:
          <Required />
        </Label>
        <br />
        <ForwardRef
          id="registration-first-name-input"
          name="first_name"
          required={true}
        />
      </div>
      <div>
        <Label
          htmlFor="registration-last-name-input"
        >
          Last name:
          <Required />
        </Label>
        <br />
        <ForwardRef
          id="registration-last-name-input"
          name="last_name"
          required={true}
        />
      </div>
      <div>
        <Label
          htmlFor="registration-username-input"
        >
          Choose a username
          <Required />
        </Label>
        <br />
        <ForwardRef
          id="registration-username-input"
          name="username"
          required={true}
        />
      </div>
      <div>
        <Label
          htmlFor="registration-password-input"
        >
          Choose a password
          <Required />
        </Label>
        <br />
        <ForwardRef
          id="registration-password-input"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <br />
      <div>
        <ForwardRef
          type="submit"
        >
          Sign up
        </ForwardRef>
      </div>
       
      <br />
      <Link
        className="already"
        to="/login"
      >
        Already have an account?
      </Link>
    </form>
  </div>
</div>
`;
